src/table/table.scss

```diff
@@ -367,6 +411,33 @@
     flex-basis: $iui-l * 2;
   }
 
+  &.iui-sticky {
+    position: sticky;
+    z-index: 1;
+  }
+
+  .iui-shadow-right,
+  .iui-shadow-left {
+    position: absolute;
+    top: -1px;
+    bottom: -1px;
+    width: 30px;
+    transition: box-shadow 0.3s;
+    pointer-events: none;
+  }
+
+  .iui-shadow-right {
+    right: 0;
+    transform: translate(100%);
+    box-shadow: inset 10px 0 8px -8px hsla(200, 12%, 0%, 0.25);
```

Comment:
Shouldn't we try to use our elevation variab;es?

Reply:
Our elevation values have the shadow go downwards where here we need them to go left / right.

Reply:
we have some random values in [`information-panel`](https://github.com/iTwin/iTwinUI/blob/main/src/information-panel/information-panel.scss#L184) too. Maybe we should align the two?

Reply:
Yeah, that would make sense.  Maybe we could make our drop shadow into a mixin that allows you to set the elevation & direction?

Reply:
Idk, our shadows feels very different and it would be too much work to make mixin because mine uses some hacks to position inset shadow outside, info-panel has parent with `overflow: hidden`, surface just has a regular outside shadow. Maybe at least we could have some variables for shadow blur.
Sticky shadow:
![image](https://user-images.githubusercontent.com/36186912/167791483-ceea5bd1-041c-4ab2-8316-c31431bbdead.png)
Info-panel:
![image](https://user-images.githubusercontent.com/36186912/167791528-bf3e2103-32cb-4af3-be05-60f6c57266b4.png)
Surface:
![image](https://user-images.githubusercontent.com/36186912/167791602-61bc482b-a6e2-48c0-a6bd-b6a65a1d7f1b.png)

Reply:
Ok, sticky cell shadow is a bit different, I tried setting the same blur value as in info-panel but because it is inset shadow I need to adjust spread and then shadows doesn't look the same. I then tried to copy whole shadow from info-panel but then it doesn't look like one shadow because there are whiter spaces:
![image](https://user-images.githubusercontent.com/36186912/168022664-35108a30-d39b-42be-8357-5da13c0a8bd6.png)
So maybe I will try to make sticky cell shadow to visually look the same as info-panel.

Reply:
I took a look at how you're implementing your shadows… pretty interesting way to make that visually work.  This screenshot shows the sticky table shadow on top & `$iui-elevation-4` but casting to the left rather than down on the bottom.  They match up pretty close.
![Screen Shot 2022-05-16 at 6 21 43 PM](https://user-images.githubusercontent.com/849817/168691872-82d20589-a08f-499b-8906-95d7ddd31044.png)

I do agree with changing `information-panel.scss` to better match our elevation, something like this for line 184 `box-shadow: -1px 0 10px hsla(200, 12%, 0%, 0.25);` which matches `$iui-elevation-4` but the X & Y swapped.

I'd like to see the sticky columns using something less elevated like `$iui-elevation-2` but that might not be achievable with the way you have the box shadow implemented.

Reply:
Tried to make sticky shadow the same as elevation-2 (top sticky, bottom one is with elevation-2):
![image](https://user-images.githubusercontent.com/36186912/169011482-2680ca06-16c6-497c-be7f-09a58947ccc8.png)
How table looks like:
![image](https://user-images.githubusercontent.com/36186912/169012441-be4833b0-3cba-4f50-b844-e5e70c8180fc.png)

Reply:
And for info panel maybe I will just create a separate PR after this one if we want to make shadows similar.